src/boot/efi/stub.c

```diff
@@ -912,6 +1017,11 @@ static EFI_STATUS run(EFI_HANDLE image) {
         generate_embedded_initrds(loaded_image, sections, initrds);
         lookup_embedded_initrds(loaded_image, sections, initrds);
 
+        /* Combine ucode initrds with the embedded initrd */
+        err = ucode_merge_and_measure_addons(initrds, &ucode_addons, &n_ucode_addons, &sections_measured);
+        if (err != EFI_SUCCESS)
+                return log_error_status(err, "Error merging ucode initrds: %m");
```

Comment:
I'd really prefer if we'd only combine initrds once instead of first combining the ucode initrds, and then combining it with the rest of the initrds.

Hence, change ucode_merge_and_measure_addons() so that it gets the fixed-size initrds[] array as input, and that it then returns a new array of initrd iovecs, which contains the ucode initrds as prefix, and then the fixed-sizes initrds[] copied into the back.

or in other words: instead of combining the actual initrd *payload* here you'd just combine the iovec array.

Reply:
aaaah right that makes sense. I tried to find an elegant solution but didn't get very far. Let me try